src/chapters/orm-relationships/many-to-many.rst

```diff
@@ -0,0 +1,690 @@
+                     Text = tag.Name
+                  });
+               }
+
+               Event = theEvent;
+         }
+
+         public AddEventTagViewModel()
+         {            
+         }
+      }
+   }
+
+This class models the data that is need to render and process our form. In order to add a tag to an event, our ``POST`` handler will need to know the IDs of the two objects in question. Therefore, our ViewModel has required ``EventId`` and ``TagId`` properties. It also contains an ``Event`` property, which we will use to display details (such as the event name) in the view.
+
+Finally, the ViewModel has a property ``List<SelectListItem> Tags``. As with previous forms containing a dropdown, this property will be used to populate the ``select`` element containing the all tag options. 
+
+The constructor requires an ``Event`` object as well as a ``List<Tag>`` object. The list will contain a collection of all tags pulled from the database when we call the constructor from within our controller.
+
+Now, let's create the view template.
+
+``Views/Tag/AddEvent.cshtml``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Our template needs a form with two inputs. The more obvious input will be the ``select`` element containing tag options. Since we also need to submit the event ID in our request, we'll add a hidden input that holds the value of ``EventId`` from our ViewModel.
+
+.. sourcecode:: html
+   :linenos:
+
+   @model CodingEventsDemo.ViewModels.AddEventTagViewModel
+
+   <h1>Add Tag to Event: @Model.Event.Name</h1>
+
+   <form asp-controller="Tag" asp-action="AddEvent" method="post">
+      <input type="hidden" value="@Model.Event.Id" name="EventId" />
+      <div class="form-group">
+         <label asp-for="TagId">Tag</label>
+         <select asp-for="TagId" asp-items="Model.Tags"></select>
+         <span asp-validation-for="TagId"></span>
+      </div>
+      <input type="submit" value="Add Tag" />
+   </form>
+
+``GET`` and ``POST`` Handlers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We're now ready to add handler methods to ``TagController``.
+
+The ``GET`` method is simple since it just displays the form.
+
+.. sourcecode:: csharp
+   :lineno-start: 50
+
+   // responds to URLs like /Tag/AddEvent/5 (where 5 is an event ID)
+   public IActionResult AddEvent(int id)
+   {
+      Event theEvent = context.Events.Find(id);
+      List<Tag> possibleTags = context.Tags.ToList();
+      AddEventTagViewModel viewModel = new AddEventTagViewModel(theEvent, possibleTags);
+      return View(viewModel);
+   }
+
+This method creates an ``AddEventTagViewModel`` using the event specified by the ``id`` parameter and the list of all tags from the database. Then it renders the view.
+
+The ``POST`` method is more complicated.
+
+.. sourcecode:: csharp
+   :lineno-start: 59
+
+   [HttpPost]
+   public IActionResult AddEvent(AddEventTagViewModel viewModel)
+   {
+      if (ModelState.IsValid)
+      {
+
+            int eventId = viewModel.EventId;
```

Comment:
trivial again but i think the indent in this block is one-too-many.

see what i did there 😉